framework_crates/bones_framework/src/networking/random.rs

```diff
@@ -0,0 +1,404 @@
+    }
+
+    /// Generate a random printable ASCII character
+    pub fn gen_random_ascii_char(&mut self) -> char {
+        self.gen_u8_range(33..=126) as char
+    }
+
+    /// Generate a random ASCII string of the specified length
+    pub fn gen_random_ascii_string(&mut self, length: u64) -> String {
+        (0..length).map(|_| self.gen_random_ascii_char()).collect()
+    }
+
+    /// Returns a boolean, where `success_rate` represents the chance to return a true value,
+    /// with 0.0 being no chance and 1.0 will always return true.
+    pub fn gen_chance(&mut self, success_rate: f64) -> bool {
+        let clamped_rate = success_rate.clamp(0.0, 1.0);
+        self.internal_generator.chance(clamped_rate)
+    }
+
+    /// Samples a random item from a slice of values.
+    pub fn gen_sample<'a, T>(&mut self, list: &'a [T]) -> Option<&'a T> {
+        self.internal_generator.sample(list)
+    }
+
+    /// Samples a random item from an iterator of values.
+    pub fn gen_sample_iter<T: Iterator>(&mut self, list: T) -> Option<T::Item> {
+        self.internal_generator.sample_iter(list)
+    }
+
+    /// Samples a random &mut item from a slice of values.
+    pub fn gen_sample_mut<'a, T>(&mut self, list: &'a mut [T]) -> Option<&'a mut T> {
+        self.internal_generator.sample_mut(list)
+    }
+
+    /// Samples multiple unique items from a slice of values.
+    pub fn gen_sample_multiple<'a, T>(&mut self, list: &'a [T], amount: usize) -> Vec<&'a T> {
+        self.internal_generator.sample_multiple(list, amount)
+    }
+
+    /// Samples multiple unique items from a mutable slice of values.
+    pub fn gen_sample_multiple_mut<'a, T>(
+        &mut self,
+        list: &'a mut [T],
+        amount: usize,
+    ) -> Vec<&'a mut T> {
+        self.internal_generator.sample_multiple_mut(list, amount)
+    }
+
+    /// Samples multiple unique items from an iterator of values.
+    pub fn gen_sample_multiple_iter<T: Iterator>(
+        &mut self,
+        list: T,
+        amount: usize,
+    ) -> Vec<T::Item> {
+        self.internal_generator.sample_multiple_iter(list, amount)
+    }
+
+    /// Stochastic Acceptance implementation of Roulette Wheel weighted selection.
+    pub fn gen_weighted_sample<'a, T, F>(
+        &mut self,
+        list: &'a [T],
+        weight_sampler: F,
+    ) -> Option<&'a T>
+    where
+        F: Fn((&T, usize)) -> f64,
+    {
+        self.internal_generator
+            .weighted_sample(list, weight_sampler)
+    }
+
+    /// Stochastic Acceptance implementation of Roulette Wheel weighted selection for mutable references.
+    pub fn gen_weighted_sample_mut<'a, T, F>(
+        &mut self,
+        list: &'a mut [T],
+        weight_sampler: F,
+    ) -> Option<&'a mut T>
+    where
+        F: Fn((&T, usize)) -> f64,
+    {
+        self.internal_generator
+            .weighted_sample_mut(list, weight_sampler)
+    }
+}
+
+fn lua_metatable(ctx: lua::Context) -> lua::Table {
```

Comment:
I think we need to gate this + the imported scripting types behind "scripting" feature flag - probably won't compile if not using default features.

(We should probably add --no-default-features CI job)